### ISSUE TITLE
Fatal error: Call to a member function compileColumnExists() on null

### ISSUE THREAD

**Issue by codingquark** [jarischaefer/hal-api#16]
When calling store on resources, I'm getting this error. I investigated and found that when calling `$this->schemaBuilder->getColumnListing($model->getTable());` in `store()` of `HalApiResourceController`, `getColumnListing` is calling `$this->grammar->compileColumnExists()`. Now, in the `Builder`, `$this->grammar` is null.

I am using postgresql. Not sure what is wrong.

**Comment by codingquark** [comment]
The same is when mysql (mariadb) is used.

**Comment by jarischaefer** [comment]
How are you instantiating the controller? Could you provide a code snippet? Perhaps the schema builder is not initialized properly.

**Comment by codingquark** [comment]
Not sure which snippet would help you understand the problem. But I'm wiring the dependencies in a service provider.

``` php
public function register()
    {
        $this->app->singleton(UsersController::class, function (\Illuminate\Contracts\Foundation\Application $application) {
            $parameters = $application->make(HalApiControllerParameters::class);
            $transformer = $application->make(UserTransformer::class);
            $schemaBuilder = $application->make(\Illuminate\Database\Schema\Builder::class);
            dd($schemaBuilder);
            return new UsersController($parameters, $transformer, $schemaBuilder);
        });

        $this->app->singleton(PostsController::class, function (\Illuminate\Contracts\Foundation\Application $application) {
            $parameters = $application->make(HalApiControllerParameters::class);
            $transformer = $application->make(PostTransformer::class);
            $schemaBuilder = $application->make(\Illuminate\Database\Schema\Builder::class);

            return new PostsController($parameters, $transformer, $schemaBuilder);
        });
```

I `dd`'d `$schemaBuilder` and in that object, `schemaGrammar` is null.

**Comment by jarischaefer** [comment]
I was able to reproduce the issue on a fresh Laravel 5.2 installation. The underlying issue is still unknown, though. I will investigate further.

**Comment by codingquark** [comment]
I looked through the code and it seems the `Builder` is not being initialized properly.

When I called `$this->connection->setSchemaGrammar(new \Illuminate\Database\Schema\Grammars\PostgresGrammar());` in Builder's constructor, things worked perfectly. This surely is not the way to go about it of course. But it gives the idea that our Builder being made by the provider is not correct.

I don't know how to correctly build the object as of now. But I'm trying to figure it out. Shall update if I come up with a fix.
